### PR TITLE
Added profile image to resolve issue #99

### PR DESCRIPTION
**Pull request by sebinfrancis7**
## Added an option to let users add profile picture if needed

Closes: #99 

## Proposed changes

An option is provided for the users to add a profile picture of their choice to the resume.

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots

https://user-images.githubusercontent.com/59263190/160250189-37bd4a7d-c14b-4262-934b-3baf97568959.mp4

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

File: src/components/Forms/Forms.js [hunk 1]
```diff
@@ -17,6 +17,12 @@ const Forms = ({ CurrentPage, SetSection, Resume }) => {
                 [fe]: e.target.value
               })
             }
+            onSwitch={() =>
+              SetSwitch()
+            }
+            ImageHandler = {e => 
+              ImageHandler(e)
+            }
```
Comment (praveenscience) [inline src/components/Forms/Forms.js:25]: Run prettier on these.

File: src/components/Forms/_FormElement.js [hunk 2]
```diff
@@ -26,11 +26,26 @@ const FormElement = ({ Label, Value, Type, onChange }) => {
...
+                  <input className="form-check-input" type="checkbox" role="switch" id="flexSwitchCheckDefault" onChange={onSwitch}/>
+                  <label className="form-check-label">Add profile picture</label>
+                </div>
+                <br/>
+                {Value.Set ? 
+                <div className="form-group">
+                  <input type="file" className="form-control-file" id="formControlFile" name="image-upload" accept="image/*" onChange={ImageHandler}/>
+                </div> 
+                : null}
+              </div>
```
Comment (praveenscience) [inline src/components/Forms/_FormElement.js:42]: Great one. Please run the prettier with the right config.

File: src/components/Main/Main.js [hunk 3]
```diff
@@ -2,14 +2,16 @@ import React from "react";
...
+const Main = ({ CurrentPage, Resume, SetSection, SetSwitch, ImageHandler }) => {
   return (
     <div className="container-fluid rezume-main">
       <div className="row">
         <div className="col-3">
           <Forms
             CurrentPage={CurrentPage}
             SetSection={SetSection}
+            SetSwitch = {SetSwitch}
+            ImageHandler = {ImageHandler}
```
Comment (praveenscience) [inline src/components/Main/Main.js:14]: No spaces around `=`.

File: src/components/Rezume/Rezume.js [hunk 4]
```diff
@@ -29,11 +29,14 @@ const Rezume = ({ Resume }) => {
...
       </div>
+      {ProfilePicture.Set ?
+      <img src = {ProfilePicture.ProfileImage} className= "img-thumbnail float-left profile-image" alt="profile"/>
+      :null }
       <h2>
         <span>{NameArray.shift()}</span> {NameArray.join(" ")}
       </h2>
       <p className="location">{Location}</p>
-      <p className="summary">
+      <p className="summary" style={{marginTop: ProfilePicture.Set ? "6rem":"2rem"}}>
```
Comment (praveenscience) [inline src/components/Rezume/Rezume.js:39]: No inline styles please.

File: src/styles/_rezume.scss [hunk 5]
```diff
@@ -38,4 +38,13 @@
...
   }
+  .profile-image{
+    position: relative;
+    top: -40px;
+    left: -30px;
+    border-radius:50%;
+    margin: 0 10px 10px 0;
+    height: 10rem;
+    width: 10rem;
+  }
```
Comment (praveenscience) [inline src/styles/_rezume.scss:49]: Run prettier on this.

**Review by praveenscience — CHANGES_REQUESTED** [review]
Really a good one. But please run prettier with the right config.

**Comment by sebinfrancis7** [comment]
> Really a good one. But please run prettier with the right config.

Do I need to make any config changes while running the Prettier extension for Vscode ?

**Comment by sebinfrancis7** [comment]
> Really a good one. But please run prettier with the right config.

Do I need to make any config changes while running the Prettier extension for Vscode ? @praveenscience

**Comment by sebinfrancis7** [comment]
> Really a good one. But please run prettier with the right config.

Do I need to make any config changes while running the Prettier extension for Vscode ? Because when I tried to use prettier, it changes the formatting of the repo code as well. @praveenscience

**Comment by praveenscience** [comment]
@sebinfrancis7 check the VS Code section of [My Personal Development Environment](https://blog.praveen.science/my-personal-development-environment/) and maybe this might help you: [My Visual Studio Code Settings · GitHub](https://gist.github.com/praveenscience/ebb5439f31774ad2fdc14cb9e7de1fc0).

**Comment by sebinfrancis7** [comment]
@praveenscience I have made the requested changes. Can you please review and merge.

**Comment by praveenscience** [comment]
Lemme check now.

File: src/components/App.js [hunk 6]
```diff
@@ -13,6 +13,10 @@ const App = () => {
         CurrentPosition: "Product Owner at Cloudroit",
         Summary:
           "Strong proficiency in JavaScript, including DOM manipulation and the JavaScript object model and thorough understanding of React JS and its core principles. Experience with popular React.js workflows (such as Flux or Redux). Familiarity with newer specifications of EcmaScript.",
+        ProfilePicture : {
+          Set : false,
+          ProfileImage : "https://cdn.nerdschalk.com/wp-content/uploads/2020/09/how-to-remove-profile-picture-on-zoom-12.png?width=150",
+        },
       },
```
Comment (praveenscience) [inline src/components/App.js:20]: Run these through prettier please.

File: src/components/Rezume/Rezume.js [hunk 7]
```diff
@@ -29,11 +29,14 @@ const Rezume = ({ Resume }) => {
...
+      {ProfilePicture.Set ?
+      <img src = {ProfilePicture.ProfileImage} className= "img-thumbnail float-left profile-image" alt="profile"/>
+      :null }
       <h2>
         <span>{NameArray.shift()}</span> {NameArray.join(" ")}
       </h2>
       <p className="location">{Location}</p>
-      <p className="summary">
+      <p className={`summary${ProfilePicture.Set ? " summary-margin":""}`}>
         <strong className="position">{CurrentPosition}</strong>
```
Comment (praveenscience) [inline src/components/Rezume/Rezume.js:40]: Run thee through prettier please.

**Review by praveenscience — CHANGES_REQUESTED** [review]

File: src/components/Rezume/Rezume.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 
 const Rezume = ({ Resume }) => {
-  const { Name, Location, CurrentPosition, Summary } = Resume.About;
+  const { Name, Location, CurrentPosition, Summary,ProfilePicture } = Resume.About;
```
Comment (praveenscience) [inline src/components/Rezume/Rezume.js:4]: Not done.

File: src/components/Rezume/Rezume.js [hunk 9]
```diff
@@ -29,14 +29,25 @@ const Rezume = ({ Resume }) => {
...
-        {Summary}
-      </p>
+      <p
+				className={`summary${
+					ProfilePicture.Set ? " summary-margin" : ""
+				}`}
+			>
+				<strong className="position">{CurrentPosition}</strong>
+				{Summary}
+			</p>
```
Comment (praveenscience) [inline src/components/Rezume/Rezume.js:50]: Prettier?

**Review by praveenscience — CHANGES_REQUESTED** [review]
How are you running the prettier on? Just run on these files alone. Should work automatically. Do you need any help?

File: src/components/Rezume/Rezume.js (hunk 9 above)
Comment (praveenscience) [reply src/components/Rezume/Rezume.js:50]: Great - I guess you just need to do one thing. Change Tabs to 2 spaces. 😊

File: src/components/Rezume/Rezume.js (hunk 8 above)
Comment (praveenscience) [reply src/components/Rezume/Rezume.js:4]: @sebinfrancis7 I really want this to be merged today. This is such a nice one. I am giving you the max points for this! <3

**Comment by sebinfrancis7** [comment]
I am doing the prettier changes manually, because while running on the whole file it changes the formatting of the code which was not written by me, which leads to the while file in git difference

File: src/components/Rezume/Rezume.js (hunk 9 above)
Comment (praveenscience) [reply src/components/Rezume/Rezume.js:50]: @sebinfrancis7 This seems fine. Just run through prettier, even if the other lines aren't your code, gets changed, it's alright, we'll tweak it later. 😊

**Comment by sebinfrancis7** [comment]
Shall I push the formatting fixes? @praveenscience

**Comment by sebinfrancis7** [comment]
Can you check? @praveenscience 
Sorry to put you through the trouble 😅.

**Review by praveenscience — APPROVED** [review]
👍

**Pull request merged**